CorsixTH/Lua/entities/patient.lua

```diff
@@ -485,24 +527,74 @@ function Patient:tapFoot()
   end
 end
 
-function Patient:goHome(cured)
+-- Increments "cures" statistics
+function Patient:countCured()
+  local hosp = self.hospital
+  hosp.num_cured = hosp.num_cured + 1
+  hosp.num_cured_ty = hosp.num_cured_ty + 1
+  local casebook = hosp.disease_casebook[self.disease.id]
+  casebook.recoveries = casebook.recoveries + 1
+  if self.is_emergency then
+    hosp.emergency.cured_emergency_patients = hosp.emergency.cured_emergency_patients + 1
+  end
+end
+
+--! Make the patient leave the hospital. This function also handles some
+--! statistics (number of cured/kicked out patients, etc.)
+--! The mood icon is updated accordingly. Reputation is impacted accordingly.
+--!param reason (string): the reason why the patient is sent home, which could be:
+--! -"cured": When the patient is cured.
+--! -"kicked": When the patient is kicked anyway, either manually,
+--! either when no treatment can be found for her/him, etc.
+--! -"over_priced": When the patient decided to leave because he/she believes
+--! the last treatment is over-priced.
+--param disease_id (string): When the reason is "over_priced" this is the
+--! id of the disease/diagnosis that the patient considered over_priced
+function Patient:goHome(reason, disease_id)
   local hosp = self.hospital
   if not hosp and self.going_home then
     -- The patient should be going home already! Anything related to the hospital
     -- will not be updated correctly, but we still want to try to get the patient to go home.
     TheApp.world:gameLog("Warning: goHome called when the patient is already going home")
-    self:setHospital(nil)
+    self:despawn()
     return
   end
-  if not cured then
+  if reason == "cured" then
+    self:setMood("cured", "activate")
+    self:changeAttribute("happiness", 0.8)
+    self.world.ui:playSound("cheer.wav") -- This sound is always heard
+    if not self.is_debug then
+      hosp:changeReputation("cured", self.disease)
+    end
+    if hosp.num_cured < 1 then
+      self.world.ui.adviser:say(_A.information.first_cure)
+    end
+    self:countCured()
+    self:updateDynamicInfo(_S.dynamic_info.patient.actions.cured)
+    self.hospital:msgCured()
```

Comment:
No, I was thinking about "msgCured", but it's not a part of this pull request anyway.